libmariadb/mariadb_lib.c

```diff
@@ -241,18 +241,30 @@ ma_net_safe_read(MYSQL *mysql)
         }
         goto restart;
       }
-      net->last_errno= last_errno;
-      if (pos[0]== '#')
+      if (last_errno >= CR_MIN_ERROR && last_errno <= CR_MAX_ERROR ||
```

Comment:
This is needed to fix travis builds, otherwise the code doesn't compile with -Werror. But I don't want to drag this any longer, I'll merge and fix afterwards with a separate commit

Reply:
> How about:
> if (IS_MYSQL_ERROR(last_errno) || IS_MARIADB_ERROR(last_errno)) ?

Per [the definition of the `IS_MYSQL_ERROR` macro](https://github.com/mariadb-corporation/mariadb-connector-c/blob/HEAD/include/errmsg.h#L124-L125), this would _not_ be equivalent.

1. Current version rejects error codes [2000, 2999] + [5000, 5999].
2. @9EOR9's proposed replacement would reject only error codes [2000, 2061] + [5000, 5023].

Because these error codes are client-side-only, they _should_ be equivalent as long as no one forgets [to update `CR_MYSQL_LAST_ERROR` and `CR_MARIADB_LAST_ERROR`](https://github.com/mariadb-corporation/mariadb-connector-c/blob/bd873535128d9e5a459def6977772d4b68934c90/include/errmsg.h#L88).

Reply:
Consider modifying the `IS_MYSQL_ERROR` and `IS_MARIADB_ERROR` macros so that they include the entire client-side error ranges, rather than depending on `CR_MYSQL_LAST_ERROR` and `CR_MARIADB_LAST_ERROR` being updated.

Reply:
That wouldn't hurt, but I'll leave it to @9EOR9. It doesn't matter much whether we test the full possible range or the actual range only, All error codes **within the range we test** can be considered safe, not injected by MitM, but generated here in the client library. All codes outside of that range are unsafe (before SSL). So as long as the code never trusts error message numbers outside of the tested range, it should be good.